src/topology/path_connected.lean

```diff
@@ -257,10 +372,104 @@ def cast (γ : path x y) {x' y'} (hx : x' = x) (hy : y' = y) : path x' y' :=
   source' := by simp [hx],
   target' := by simp [hy] }
 
+@[simp] lemma symm_cast {X : Type*} [topological_space X] {a₁ a₂ b₁ b₂ : X}
+  (γ : path a₂ b₂) (ha : a₁ = a₂) (hb : b₁ = b₂) :
+  (γ.cast ha hb).symm = (γ.symm).cast hb ha := rfl
+
+@[simp] lemma trans_cast {X : Type*} [topological_space X] {a₁ a₂ b₁ b₂ c₁ c₂ : X}
+  (γ : path a₂ b₂) (γ' : path b₂ c₂) (ha : a₁ = a₂) (hb : b₁ = b₂) (hc : c₁ = c₂) :
+  (γ.cast ha hb).trans (γ'.cast hb hc) = (γ.trans γ').cast ha hc := rfl
+
 @[simp] lemma cast_coe (γ : path x y) {x' y'} (hx : x' = x) (hy : y' = y) :
   (γ.cast hx hy : I → X) = γ :=
 rfl
 
+lemma symm_continuous_family {X ι : Type*} [topological_space X] [topological_space ι]
+  {a b : ι → X} (γ : Π (t : ι), path (a t) (b t)) (h : continuous ↿γ) :
+  continuous ↿(λ t, (γ t).symm) :=
+h.comp (continuous_id.prod_map continuous_I_symm)
+
+lemma continuous_uncurry_extend_of_continuous_family {X ι : Type*} [topological_space X]
+  [topological_space ι] {a b : ι → X}  (γ : Π (t : ι), path (a t) (b t)) (h : continuous ↿γ) :
+  continuous ↿(λ t, (γ t).extend) :=
+h.comp (continuous_id.prod_map continuous_proj_I)
+
+lemma trans_continuous_family {X ι : Type*} [topological_space X] [topological_space ι]
+  {a b c : ι → X}
+  (γ₁ : Π (t : ι), path (a t) (b t)) (h₁ : continuous ↿γ₁)
+  (γ₂ : Π (t : ι), path (b t) (c t)) (h₂ : continuous ↿γ₂) :
+  continuous ↿(λ t, (γ₁ t).trans (γ₂ t)) :=
+begin
+  have h₁' := path.continuous_uncurry_extend_of_continuous_family γ₁ h₁,
+  have h₂' := path.continuous_uncurry_extend_of_continuous_family γ₂ h₂,
+  simp [has_uncurry.uncurry, has_coe_to_fun.coe, coe_fn, path.trans],
+  apply continuous_if _ _ _,
+  { rintros st hst,
+    have := frontier_le_subset_eq (continuous_subtype_coe.comp continuous_snd)
+      continuous_const hst,
+    simp only [mem_set_of_eq, comp_app] at this,
+    simp [this, mul_inv_cancel (@two_ne_zero ℝ _)] },
+  { change continuous ((λ p : ι × ℝ, (γ₁ p.1).extend p.2) ∘ (prod.map id (λ x, 2*x : I → ℝ))),
+    exact h₁'.comp (continuous_id.prod_map $ continuous_const.mul continuous_subtype_coe) },
+  { change continuous ((λ p : ι × ℝ, (γ₂ p.1).extend p.2) ∘ (prod.map id (λ x, 2*x - 1 : I → ℝ))),
+    exact h₂'.comp (continuous_id.prod_map $
+      (continuous_const.mul continuous_subtype_coe).sub continuous_const) },
+end
+
+/-! #### Portion of a path -/
+
+/-- `γ.portion t` is the path which starts by following the path `γ`,
```

Comment:
I don't like this definition, because it is somewhat arbitrary that you only keep the beginning of the path. You could have `path.truncate_left`, `path.truncate_right` and `path.truncate`, for instance (or just `path.truncate`, where you allow truncation on both sides, and where you could use `0` on the left if you want to copy the functionality of your "portion")

Reply:
I don't think it's so arbitrary, because the path is clearly oriented, and I think truncating means keeping the beginning.

Reply:
To be honest I was also not completely okay with my `portion`, so I just made your `truncate`. I'll have to modify my proof in the sphere eversion project but I've made a few tests and it should hopefully not be much longer.